src/Compilers/CSharp/Portable/Symbols/ArrayTypeSymbol.cs

```diff
@@ -390,35 +395,20 @@ internal override bool ApplyNullableTransforms(ImmutableArray<bool> transforms,
                 return false; 
             }
 
-            if (oldElementType.IsSameAs(newElementType))
-            {
-                result = this;
-            }
-            else
-            {
-                result = IsSZArray ?
-                    ArrayTypeSymbol.CreateSZArray(newElementType, _baseType) :
-                    ArrayTypeSymbol.CreateMDArray(newElementType, Rank, Sizes, LowerBounds, _baseType);
-            }
-
+            result = WithElementType(newElementType);
             return true;
         }
 
         internal override TypeSymbol SetUnknownNullabilityForReferenceTypes()
         {
-            TypeSymbolWithAnnotations oldElementType = ElementType;
-            TypeSymbolWithAnnotations newElementType = oldElementType.SetUnknownNullabilityForReferenceTypes();
+            return WithElementType(ElementType.SetUnknownNullabilityForReferenceTypes());
+        }
 
-            if (oldElementType.IsSameAs(newElementType))
-            {
-                return this;
-            }
-            else
-            {
-                return IsSZArray ?
-                    ArrayTypeSymbol.CreateSZArray(newElementType, _baseType) :
-                    ArrayTypeSymbol.CreateMDArray(newElementType, Rank, Sizes, LowerBounds, _baseType);
-            }
+        internal override TypeSymbol MergeNullability(TypeSymbol other, VarianceKind variance, out bool hadNullabilityMismatch)
+        {
+            Debug.Assert(this.Equals(other, TypeCompareKind.IgnoreDynamicAndTupleNames));
+            TypeSymbolWithAnnotations elementType = ElementType.MergeNullability(((ArrayTypeSymbol)other).ElementType, VarianceKind.None, out hadNullabilityMismatch);
```

Comment:
>None [](start = 132, length = 4)

Arrays of reference type are covariant. Having said that, I think this is correct.